src/interfaces/gateway/IAggregator.sol

```diff
@@ -2,20 +2,25 @@
 pragma solidity >=0.5.0;
 
 interface IAggregator {
+    /// @dev Each router struct is packed with the quorum to reduce SLOADs on handle
     struct Router {
-        // Starts at 1 and maps to id - 1 as the index on the routers array
+        /// @notice Starts at 1 and maps to id - 1 as the index on the routers array
         uint8 id;
-        // Each router struct is packed with the quorum to reduce SLOADs on handle
+        /// @notice Number of votes required for a message to be executed
         uint8 quorum;
+        /// @notice Each time routers are updated, a new session starts which invalidates old votes
+        uint64 activeSessionId;
     }
 
-    struct ConfirmationState {
-        // Counts are stored as integers (instead of boolean values) to accommodate duplicate
-        // messages (e.g. two investments from the same user with the same amount) being
-        // processed in parallel. The entire struct is packed in a single bytes32 slot.
-        // Max uint16 = 65,535 so at most 65,535 duplicate messages can be processed in parallel.
-        uint16[8] messages;
-        uint16[8] proofs;
+    struct Message {
+        /// @dev Counts are stored as integers (instead of boolean values) to accommodate duplicate
+        ///      messages (e.g. two investments from the same user with the same amount) being
+        ///      processed in parallel. The entire struct is packed in a single bytes32 slot.
+        ///      Max uint16 = 65,535 so at most 65,535 duplicate messages can be processed in parallel.
+        uint16[8] votes;
+        /// @notice Each time routers are updated, a new session starts which invalidates old votes
```

Comment:
Actually that's not always true. A new sessions starts only if the new routers count ( quorum ) is less than the current one, isn't this correct?